lib/builders.nix

```diff
@@ -0,0 +1,31 @@
+{
+  lib,
+  pkgs,
+}: {
+  /*
+  Write a lua file to the nix store, formatted using stylua.
+
+  # Type
+
+  ```
+  writeLua :: String -> AttrSet -> String -> Derivation
+  ```
+
+  # Arguments
+
+  - [name] The name of the derivation
+  - [text] The content of the lua file
+  */
+  writeLua = name: text: let
+    raw = pkgs.writeText name text;
+    stylua = lib.getExe pkgs.stylua;
+  in
+    pkgs.runCommand name {} ''
```

Comment:
Couldn't we use the env vars to pass the raw text, and use ` <<< "$luaFile` instead of creating an unformatted text file in the store?

Reply:
Was just about to ask if `lib.escapeShellArg` was preferable to writing an extra file to the store. I take it you think so?

I figured it added extra overhead escaping a large/complex file, but writing redundant stuff to the store isn't ideal either.

Reply:
Env vars shold avoide the neeed to esacape

Reply:
Oh you mean the env vars attrset passed to `runCommand`, good idea